### ISSUE TITLE
NullReferenceException when controlling a turret.

### ISSUE THREAD

**Issue by SomeRandomDude-git** [sstixrud/WeaponCore#555]
## Is this a multiplayer or singleplayer bug?

Singleplayer Bug & Locally hosted multiplayer.
I have not tested it on a dedicated server.

## Describe the bug:

When sitting in a control seat and controlling a turret, the turret will not shoot when the player clicks. A NRE is printed in the debug log. I have also fiddled around with the various firing modes to no avail.

```
06-20-23_11-21-22-231 - Exception in OnPlayerController: System.NullReferenceException: Object reference not set to an instance of an object.
   at CoreSystems.Session.GunnerAcquire(MyEntity entity, Int64 playerId)
   at CoreSystems.Platform.Weapon.WeaponComponent.TookControl(Int64 playerId)
   at CoreSystems.Session.OnPlayerControl(MyEntity exitEntity, MyEntity enterEntity)
06-20-23_11-21-28-327 - Exception in OnPlayerController: System.NullReferenceException: Object reference not set to an instance of an object.
   at CoreSystems.Session.GunnerRelease(Int64 playerId)
   at CoreSystems.Platform.Weapon.WeaponComponent.ReleaseControl(Int64 playerId)
   at CoreSystems.Session.OnPlayerControl(MyEntity exitEntity, MyEntity enterEntity)
```

## Steps To Reproduce:

Loaded into a new singleplayer empty world on creative mode. Made a new grid with a control seat, reactor, and a Gatling turret, and was unable to shoot when controlling it. Also tested in survival SP.

## Misc:

✅ Followed the instructions to clear the cache.
✅ Verified integrity of game files.
✅ Only WeaponCore loaded.
✅ No blueprints used, made on a fresh grid.
✅ Also tested with other WeaponCore mods, returns the same NRE.

## The logs are here:
[1918681825.sbm_CoreSystems.zip](https://github.com/sstixrud/WeaponCore/files/11805868/1918681825.sbm_CoreSystems.zip)

**Comment by sstixrud** [comment]
I am not able to reproduce this with a clean world, with a new grid and controlling the gatling turret.

**Comment by SomeRandomDude-git** [comment]
Hmm, that's odd. Ill do a fresh install of Space engineers to see if that fixes it.

**Comment by SomeRandomDude-git** [comment]
Reinstalling SE didn't seem to work, but after an hour of trial and error testing the only thing that seems to fix the issue is by deleting CoreSystemsClient.Cfg in appdata\Roaming\SpaceEngineers\Storage . For whatever reason deleting that file fixes the issue, but once I quit the session and load the save again / make a new world, the error returns until I delete it again after it regenerates. I'm sure the issue isn't directly caused by a config file, as that makes no sense, but I honestly have no idea what the actual root problem is.

I know all that isn't really helpful. It doesn't seem to be a permissions issue as it doesn't seem to have any issue writing data to the .cfg when I use the ingame /wc commands, nor does running SE as an administrator have any effect. Ill keep chipping away at it and see if I can figure out whats causing it.

Juuuust to be safe, heres the .cfg file thats being generated for me:

```
<?xml version="1.0" encoding="utf-16"?>
<ClientSettings xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
  <Version>9</Version>
  <ClientOptimizations>true</ClientOptimizations>
  <AvLimit>0</AvLimit>
  <MenuButton>Middle</MenuButton>
  <ControlKey>R</ControlKey>
  <ShowHudTargetSizes>false</ShowHudTargetSizes>
  <ActionKey>NumPad0</ActionKey>
  <HudPos>
    <X>0</X>
    <Y>0</Y>
  </HudPos>
  <HudScale>1</HudScale>
  <InfoKey>Decimal</InfoKey>
  <MinimalHud>false</MinimalHud>
  <StikcyPainter>true</StikcyPainter>
  <CycleNextKey>PageDown</CycleNextKey>
  <CyclePrevKey>PageUp</CyclePrevKey>
  <AdvancedMode>false</AdvancedMode>
  <HideReload>false</HideReload>
</ClientSettings>
```

**Comment by SomeRandomDude-git** [comment]
Ok, I've now found a permanent solution. Deleting my main SpaceEngineers.cfg and letting it regenerate has fixed the issue for me. After taking a look at my old file and the new one I noticed there's tons of entries in the old cfg that just aren't used in the new one, Perhaps due to it being originally created back in 2019. I'm sure more than an error or two were being caused by this.

Apologies if I wasted any of your time, you would think that SE cleans up its own configs after updates... 🤔 

Here's the old config if its relevant at all, but other than that that solves the issue.
[SpaceEngineersConfig.zip](https://github.com/sstixrud/WeaponCore/files/11824386/SpaceEngineersConfig.zip)
